tests/rptest/remote_scripts/topic_operations.py

```diff
@@ -0,0 +1,397 @@
+            # Prepare next batch
+            topic_batch = []
+            remaining_count = target_count - count_created
+            next_batch_size = remaining_count if remaining_count < batch_size else batch_size
+            for i in range(next_batch_size):
+                index = count_created + i
+                _topic = deepcopy(_topic_item)
+                # This is for code cleaniness
+                # since the order will be differnt after ThreadPoolExecutor
+                _topic['index'] = count_created + i
+                # Since new batch is already started this is always +1
+                _topic['batch-index'] = index // batch_size + 1
+                _topic['name'] = topic_names[index]
+                _topic['partitions'] = num_partitions
+                _topic['replicas'] = num_replicas
+                topic_batch.append(_topic)
+
+            # Create batch
+            batch_start_s = time.time()
+            if use_kafka_batch:
+                # Topic creation
+                self.logger.info(f"Creating {target_count} topics using "
+                                 f"batch size of {batch_size}")
+                # If kafka native batching used, just send it all
+                created_topics = _create_topic_batch(topic_batch)
+                topics += created_topics
+            else:
+                # Use pool executor with worker count and send per-topic
+                # creation requests
+                self.logger.info(f"Creating {target_count} topics: "
+                                 f"batch size = {batch_size}, "
+                                 f"pool workers = {workers}")
+                with ThreadPoolExecutor(max_workers=workers) as executor:
+                    topic_index = count_created + 1
+                    for result in executor.map(_create_single_topic,
+                                               topic_batch):
+                        # Set topic index based on order of creation
+                        result['index'] = topic_index
+                        topics.append(result)
+                        topic_index += 1
+            batch_creation_time_s = time.time() - batch_start_s
+
+            # Keep counting
+            count_created += batch_size
+
+            # Peek creation timings
+            timings['count_created'] = count_created
+            timings['batch_timings_s'].append(batch_creation_time_s)
+            # min/max
+            creation_times = [t["time-to-create-sec"] for t in topics]
+            timings["creation-time-max"] = max(creation_times)
+            timings["creation-time-min"] = min(creation_times)
+            timings['creation_times'] = creation_times
+
+            # TODO: Add json output for ongoing data
+            write_json(sys.stdout, {'timings': timings})
+            # Exit on threshold > 5 min
+            # I.e. single topic creation takes more than 5 min
+            if timings["creation-time-max"] > 300:
```

Comment:
nit: It would be nicer to proactively timeout after 5mins. Rather than this approach of waiting however long topic creation(for a batch or single topic) takes then erroring out if it takes longer then 5min. It's something that can be done in a follow up PR though

Reply:
Agreed. Will update.